llama/model.py

```diff
@@ -87,100 +95,114 @@ def repeat_kv(x: torch.Tensor, n_rep: int) -> torch.Tensor:
 
 
 class Attention(nn.Module):
-    def __init__(self, args: ModelArgs):
+    def __init__(self, args: ModelArgs, world_size: Optional[int] = None,
+                 rank: Optional[int] = None, groups: Optional[List] = None):
         super().__init__()
-        self.n_kv_heads = args.n_heads if args.n_kv_heads is None else args.n_kv_heads
-        model_parallel_size = fs_init.get_model_parallel_world_size()
-        self.n_local_heads = args.n_heads // model_parallel_size
-        self.n_local_kv_heads = self.n_kv_heads // model_parallel_size
-        self.n_rep = self.n_local_heads // self.n_local_kv_heads
+        #NEW ADDITION
+        #self.n_kv_heads = args.n_heads if args.n_kv_heads is None else args.n_kv_heads
+        #model_parallel_size = fs_init.get_model_parallel_world_size()
+        #self.n_local_heads = args.n_heads // model_parallel_size
+        #self.n_local_kv_heads = self.n_kv_heads // model_parallel_size
```

Comment:
"grouped-query attention" is the major LLaMA 2 difference from LLaMA 1. Without it, there is no need to adapt optimization to LLaMA 2, because the code would be same.

Reply:
the grouped-query attention is purposely commented - as discussed let's address that in the follow up PR.